web/client-api/src/main/java/io/deephaven/web/client/api/parse/JsDataHandler.java

```diff
@@ -0,0 +1,488 @@
+                JsConsumer<Uint8Array> addBuffer) {
+            writeSimpleNumbers(data, addNode, addBuffer, Float64Array.BYTES_PER_ELEMENT, NULL_DOUBLE,
+                    Float64Array::new);
+        }
+    },
+    BOOLEAN(Type.Int, "boolean") {
+        @Override
+        public double writeType(Builder builder) {
+            return Int.createInt(builder, 8, true);
+        }
+
+        @Override
+        public void write(Object[] data, ParseContext context, JsConsumer<Node> addNode,
+                JsConsumer<Uint8Array> addBuffer) {
+            int nullCount = 0;
+            BitSet nulls = new BitSet(data.length);
+            Int8Array payload = makeBuffer(data.length, Int8Array.BYTES_PER_ELEMENT, Int8Array::new);
+            for (int i = 0; i < data.length; i++) {
+                Object val = data[i];
+                byte boolValue;
+                if (val == null) {
+                    boolValue = NULL_BOOLEAN_AS_BYTE;
+                } else {
+                    String typeof = Js.typeof(val);
+                    switch (typeof) {
+                        case "boolean":
+                            boolValue = Js.isTruthy(val) ? TRUE_BOOLEAN_AS_BYTE : FALSE_BOOLEAN_AS_BYTE;
+                            break;
+                        case "number":
+                            boolValue = Js.asByte(val);// checks the values to ensure it is a byte
+                            break;
+                        case "string":
+                            String str = Js.asString(val);
+                            switch (str.toLowerCase()) {
+                                case "true":
+                                    boolValue = TRUE_BOOLEAN_AS_BYTE;
+                                    break;
+                                case "false":
+                                    boolValue = FALSE_BOOLEAN_AS_BYTE;
+                                    break;
+                                case "null":
+                                    boolValue = NULL_BOOLEAN_AS_BYTE;
+                                    break;
+                                default:
+                                    boolValue = Byte.parseByte(str);
+                                    break;
+                            }
+                            break;
+                        default:
+                            throw new IllegalArgumentException(
+                                    "Unsupported type to handle as a boolean value " + typeof);
+                    }
+                }
+
+                if (boolValue != FALSE_BOOLEAN_AS_BYTE && boolValue != TRUE_BOOLEAN_AS_BYTE
+                        && boolValue != NULL_BOOLEAN_AS_BYTE) {
+                    throw new IllegalArgumentException("Can't handle " + val + " as a boolean value");
+                }
+
+                // write the value, and mark non-null if necessary
+                if (boolValue != NULL_BOOLEAN_AS_BYTE) {
+                    nulls.set(i);
```

Comment:
this is inconsistent w.r.t. a corresponding nullCount -- however, I think this use case also applies; that we explicitly want to send the empty validity buffer.

Reply:
At this time, I am not sending a dhNulls=true, but doing a vanilla DoPut, just with deephaven:type metadata.